src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rerank/validate.ts

```diff
@@ -20,35 +32,72 @@ export const validate = (
 ): ESQLMessage[] => {
   const messages: ESQLMessage[] = [];
 
-  // Run standard argument validation
-  messages.push(...validateCommandArguments(command, ast, context, callbacks));
+  const { query, targetField, location } = command as ESQLAstRerankCommand;
+
+  // Sets the target field so the column is recognized after the command is applied
+  const targetName = targetField?.name || 'rerank';
+
+  context?.userDefinedColumns.set(targetName, [
+    {
+      name: targetName,
+      location: targetField?.location || location,
+      type: 'keyword',
+    },
+  ]);
 
-  // Cast to RERANK command for type-specific validation
-  const rerankCommand = command as ESQLAstRerankCommand;
+  const rerankExpressionType = getExpressionType(
+    query,
+    context?.fields,
+    context?.userDefinedColumns
+  );
 
-  // Validate that query text is provided
-  if (!rerankCommand.query) {
+  // check for supported query types
+  if (!supportedQueryTypes.includes(rerankExpressionType)) {
     messages.push({
-      location: command.location,
-      text: i18n.translate('kbn-esql-ast.esql.validation.rerankMissingQuery', {
-        defaultMessage: '[RERANK] Query text is required.',
+      location: 'location' in query ? query?.location : location,
+      text: i18n.translate('kbn-esql-ast.esql.validation.rerankUnsupportedFieldType', {
+        defaultMessage: '[RERANK] query must be of type [text] but is [{rerankExpressionType}]',
+        values: { rerankExpressionType },
       }),
       type: 'error',
-      code: 'rerankMissingQuery',
+      code: 'rerankUnsupportedFieldType',
     });
   }
 
-  // Validate that at least one field is specified in ON clause
-  if (!rerankCommand.fields || rerankCommand.fields.length === 0) {
-    messages.push({
-      location: command.location,
-      text: i18n.translate('kbn-esql-ast.esql.validation.rerankMissingFields', {
-        defaultMessage: '[RERANK] At least one field must be specified in the ON clause.',
-      }),
-      type: 'error',
-      code: 'rerankMissingFields',
-    });
+  // check for WITH clause and require inference_id
+  const withOption = command.args.find(
+    (arg): arg is ESQLCommandOption => isOptionNode(arg) && arg.name === 'with'
+  );
+  const hasWith = !!withOption || (command.text || '').toLowerCase().includes('with');
+
+  if (hasWith) {
+    const map = (withOption?.args?.[0] as ESQLMap | undefined) || undefined;
+
+    const key =
+      map && isMap(map) && !map.incomplete
+        ? map.entries?.find((entry) => WITH_OPTION_NAMES.includes(entry.key?.valueUnquoted))?.value
+        : undefined;
+
+    const isEmptyString = !!(
+      key &&
+      isLiteral(key) &&
+      key.literalType === 'keyword' &&
+      (key.valueUnquoted?.length ?? 0) === 0
+    );
+
+    if (!key || key.incomplete || isEmptyString) {
+      messages.push({
```

Comment:
Can we add this message to `src/platform/packages/shared/kbn-esql-ast/src/definitions/utils/errors.ts` instead of building directly here? That is intended to be our comprehensive list of validation errors.